pyvista/utilities/geometric_objects.py

```diff
@@ -1218,6 +1219,77 @@ def Circle(radius=0.5, resolution=100):
     return pyvista.wrap(pyvista.PolyData(points, cells))
 
 
+def Superquadric(center=(0., 0., 0.), scale=(1., 1., 1.), size=0.5,
+                 theta_roundness=1., phi_roundness=1.,
+                 theta_resolution=16, phi_resolution=16,
+                 toroidal=False, thickness=1/3):
+    """Create a superquadric.
+
+    Parameters
+    ----------
+    center : iterable, optional
+        Center of the superquadric in ``[x, y, z]``.
+
+    scale : iterable, optional
+        Scale factors of the superquadric in ``[x, y, z]``.
+
+    size : float, optional
+        Superquadric isotropic size.
+
+    theta_roundness : float, optional
+        Superquadric east/west roundness.
+        Values range from 0 (rectangular) to 1 (circular) to higher orders.
+
+    phi_roundness : float, optional
+        Superquadric north/south roundness.
+        Values range from 0 (rectangular) to 1 (circular) to higher orders.
+
+    theta_resolution : int, optional
+        Number of points in the longitude direction.
+        Values are rounded to nearest multiple of 4.
+
+    phi_resolution : int, optional
+        Number of points in the latitude direction.
+        Values are rounded to nearest multiple of 8.
+
+    toroidal : bool, optional
+        Whether or not the superquadric is toroidal (1) or ellipsoidal (0).
+
+    thickness : float, optional
+        Superquadric ring thickness.
+        Only applies if toroidal is set to 1.
+
+    Returns
+    -------
+    pyvista.PolyData
+        Superquadric mesh.
+
+    See Also
+    --------
+    pyvista.ParametricSuperEllipsoid : Parametric superquadric if toroidal is 0.
+    pyvista.ParametricSuperToroid : Parametric superquadric if toroidal is 1.
+
+    Examples
+    --------
+    >>> import pyvista
+    >>> superquadric = pyvista.Superquadric()
+    >>> superquadric.plot(show_edges=True)
```

Comment:
This example in the docstring gives an uninteresting sphere object that does not showcase the superquadric capabilities.  I would suggest a more interesting parameter choice for the documentation.

![image](https://user-images.githubusercontent.com/39341281/143324544-e3c1ff2c-789b-4e99-b9b6-6d55c1f47ba8.png)

Reply:
I've chosen for `scale=(3., 1., 0.5)` and roundnesses `0.1` and `0.5`. This gives:

<img width="496" alt="image" src="https://user-images.githubusercontent.com/13775259/143473959-06ce3563-015f-4013-a32f-539e65d3a9dc.png">